model/patch/patch.go

```diff
@@ -808,6 +808,21 @@ func (p *Patch) GetPatchIndex(parentPatch *Patch) (int, error) {
 	return -1, nil
 }
 
+// GetGithubContextForChildPatch returns the github context for the given child patch, to be used in github statuses.
+func GetGithubContextForChildPatch(projectIdentifier string, parentPatch, childPatch *Patch) (string, error) {
+	var githubContext string
+	patchIndex, err := childPatch.GetPatchIndex(parentPatch)
+	if err != nil {
+		return "", errors.Wrap(err, "getting child patch index")
+	}
+	if patchIndex == 0 || patchIndex == -1 {
```

Comment:
Can we add a small comment on why this if/else block is necessary?

Reply:
I'm honestly not sure since I grabbed this from elsewhere; I'm assuming it's because the patch might not have been created yet? Or to catch the case that it's not actually being called with a child patch? cc @malikchaya2 can you weigh in

Reply:
We needed a way to identify child patches uniquely so that Github can update the status. Using `evergreen/projectIdentifier` wasn't enough because there can be two child patches in the same project. We use the index as a unique key. 
We don't need it if the index is -1 because that is used to indicate that the patch is not a child patch. 0 doesn't need it because it can just be `"evergreen/project"` instead of `"evergreen/project/index"` and still be unique. 

If we want to, we can add a comment to the effect of: 
"if the patch is a child patch and it's not the only child, use the index as a unique identifier to disambiguate the child patches". 

I'm also realizing that we can remove the -1 altogether because it checks to make sure that the patch has a parent before getting the index. I created https://jira.mongodb.org/browse/EVG-19081 to clean it up, so I can add the comment then.

Reply:
by "in the same project" do you mean two child patches within the same child patch project?

Reply:
yes

Reply:
Ah ok, I understand - can we add a comment on this line consolidating this explanation?